lib/stats-collector.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+/**
+ * Stats Collector
+ *
+ * Provides stats such as test duration,
+ * number of tests passed / failed etc.
+ *
+ * @public
+ * @api public
+ * @memberof Mocha
+ * @param {Runner} runner
+ * @return {stats}
+ */
+function statsCollector(runner) {
+  var stats = {
+    suites: 0,
+    tests: 0,
+    passes: 0,
+    pending: 0,
+    failures: 0
+  };
+  if (!runner) {
+    throw new Error('Missing runner argument');
+  }
+
+  runner.stats = stats;
+
+  runner.once('start', function() {
+    stats.start = new Date();
+  });
+
+  runner.on('suite', function(suite) {
+    stats.suites = stats.suites || 0;
+    suite.root || stats.suites++;
+  });
+
+  runner.on('test end', function() {
```

Comment:
Move to before `runner.once('end', ` such that events listed in order of occurrence.

Reply:
@boneskull, why does Mocha count "suites" as soon as they start, yet count "tests"  only when they complete?